### PR TITLE
This PR fully removes the deprecated styling system and moves us over to functional components. 

### PR DESCRIPTION
**Pull request by obra**
Along the way, it refactors the layout sharing / backup&restore screens into multiple files to make them easier to understand and maintain.

### CODE REVIEW

**Review by algernon — CHANGES_REQUESTED** [review]
The code looks good, I like it.

However, I noticed a couple of issues while testing:

- The SaveChangesButton is directly at the bottom of the screen, it is missing the margin it used to have. That's easy to fix, just wrap it in a `<Box sx={{ mb: 1 }}>` rather than a div with a className that no longer does anything.
- The Model01 - and likely the Model100 too - fails to render. I'll include the error later.

**Comment by algernon** [comment]
Fixed the Model01/Model100 issue (it was a stray `this.`), but the SaveChangesButton is completely borked.

**Comment by obra** [comment]
Ack. Will fix later today.

File: src/renderer/components/SaveChangesButton.js [hunk 1]
```diff
@@ -17,154 +17,95 @@
...
+      </Button>
+    </Box>
+  );
 
-    const icon = this.props.icon || <SaveAltIcon />;
+  const icon = props.icon || <SaveAltIcon />;
 
-    const OptionalTooltip = (props) => {
-      if (this.props.floating) {
-        return <Tooltip title={this.props.children}>{props.children}</Tooltip>;
```
Comment (algernon) [inline src/renderer/components/SaveChangesButton.js:131]: This is one place where the component breaks: in the old code, we had `props` (local to `OptionalTooltip`) and `this.props` (from `SaveChangesButton`). We need to treat those separately.

We probably want to `const { children } = props` outside of `OptionalTooltip`, and use `children` and `props` children in there. Or just rename the `props` argument of `OptionalTooltip`.

File: src/renderer/components/SaveChangesButton.js [hunk 2]
```diff
@@ -17,154 +17,95 @@
...
+  return (
+    <OptionalTooltip>
+      <Box
+        sx={{
+          position: "fixed",
+          justifyContent: "flex-end",
+          bottom: 0,
+          left: 32,
+          zIndex: (theme) => theme.zIndex.drawer + 1,
+        }}
```
Comment (algernon) [inline src/renderer/components/SaveChangesButton.js:82]: These are properties we need to add *only* when `props.floating` is set.

We're using the `SaveChangesButton` both in the Editor, as a Floating Action Button, and at other places, like the `Upgrade` button on the FirmwareUpgrade screen, where it is a circular button + a normal button kinda mashed together.

We broke the opacity styling at some point, so they no longer look that well....

On second thought, see my global comment regarding this component. I'll post that in a minute.

**Comment by algernon** [comment]
So, the `SaveChangesButton` component is an ugly monster, and I'm sorry for bringing it alive.

The intent was to provide a single component that can present a button that gives feedback about the save progress itself, so we can use it whenever we're doing a longer operation. At its core, it's a circular Fab button, which has a green circle swirling around it while the save is in progress, and then turns into a success button briefly when done. On top of that, it has an extra feature that in a non-floating context, it's a circular Fab + a normal button mashed together.

We should probably rethink that, the Editor is the only place where we use the `floating` variant, so it could just... do its own thing, while other places that use `SaveChangesButton` could do the non-floating variant, probably without a circular button, and the progress swirling around the normal button, or some other kind of feedback.

File: src/renderer/components/SaveChangesButton.js (hunk 2 above)
Comment (obra) [reply src/renderer/components/SaveChangesButton.js:84]: I think this is addressed now. I agree that eventually we should split this apart, but for now this should be ok.

**Review by algernon — APPROVED** [review]

**Pull request merged**